api/pkg/s3/api-response.go

```diff
@@ -194,4 +194,37 @@ func parseListUploadsQuery(query url.Values) (request ListUploadsRequest, err er
 	request.Prefix = query.Get("prefix")
 	request.UploadIdMarker = query.Get("upload-id-marker")
 	return
+}
+
+// Parse object url queries
+func parseListObjectPartsQuery(query url.Values) (request ListPartsRequest, err error) {
+	request.EncodingType = query.Get("encoding-type")
+	request.UploadId = query.Get("uploadId")
+	if request.UploadId == "" {
+		err = ErrNoSuchUpload
+		return
+	}
+	if query.Get("max-parts") == "" {		request.MaxParts = MaxPartsList
+	} else {
+		request.MaxParts, err = strconv.Atoi(query.Get("max-parts"))
+		if err != nil {
+			return
```

Comment:
Could you please add a log for each error return?

Reply:
@sfzeng ,  when parseListObjectPartsQuery is called,  the error log is printed at caller. Is it necessary to print log here?

Reply:
I just thought that would be better for debug.

Reply:
ok, will add log here, thanks